### PR TITLE
add nodepools and zones support for CLI

### PR DESCRIPTION
**Pull request by adilliadil**
---

This checklist is used to make sure that common guidelines for a pull request are followed.

- [x] The PR has modified HISTORY.rst describing any customer-facing, functional changes. Note that this does not include changes only to help content. (see [Modifying change log](https://github.com/Azure/azure-cli/tree/master/doc/authoring_command_modules#modify-change-log)).

- [x] I adhere to the [Command Guidelines](https://github.com/Azure/azure-cli/blob/dev/doc/command_guidelines.md).

### CODE REVIEW

**Comment by adilliadil** [comment]
@xizhamsft

**Comment by yonzhan** [comment]
@qwordy  please take a look at this PR.

File: src/azure-cli/azure/cli/command_modules/acs/_params.py [hunk 1]
```diff
@@ -176,6 +176,7 @@ def load_arguments(self, _):
         c.argument('load_balancer_outbound_ips', type=str, validator=validate_load_balancer_outbound_ips)
         c.argument('load_balancer_outbound_ip_prefixes', type=str, validator=validate_load_balancer_outbound_ip_prefixes)
         c.argument('vm_set_type', type=str, validator=validate_vm_set_type)
+        c.argument('zones', zones_type, options_list='--zones', help='Space-separated list of availability zones where agent nodes will be placed.')
```
Comment (qwordy) [inline src/azure-cli/azure/cli/command_modules/acs/_params.py]: Maybe you can add an alias `-z` just like `vm/vmss create` to provide consistent experience across commands.

File: src/azure-cli/azure/cli/command_modules/acs/_params.py [hunk 2]
```diff
@@ -257,6 +258,28 @@ def load_arguments(self, _):
         c.argument('nodepool_name', type=str,
                    help='Node pool name, upto 12 alphanumeric characters', validator=validate_nodepool_name)
 
+    with self.argument_context('aks nodepool') as c:
+        c.argument('cluster_name', type=str, help='The cluster name.')
+
+    for scope in ['aks nodepool add']:
+        with self.argument_context(scope) as c:
+            c.argument('nodepool_name', type=str, options_list=['--name', '-n'], validator=validate_nodepool_name, help='The node pool name.')
+            c.argument('zones', zones_type, options_list='--zones', help='Space-separated list of availability zones where agent nodes will be placed.')
```
Comment (qwordy) [inline src/azure-cli/azure/cli/command_modules/acs/_params.py]: Maybe you can add an alias `-z` just like `vm/vmss create` to provide consistent experience across commands.

Comment (adilliadil) [reply src/azure-cli/azure/cli/command_modules/acs/_params.py]: Done

File: src/azure-cli/azure/cli/command_modules/acs/_params.py (hunk 1 above)
Comment (adilliadil) [reply src/azure-cli/azure/cli/command_modules/acs/_params.py]: Done

**Comment by qwordy** [comment]
Help-Entry: `aks nodepool add` - The following parameter help names are invalid: --zones

**Pull request merged**